libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+      (val) =>
+        createDescription(
+          val as Record<string, unknown>,
+          _.get(temp, 'result.score') as number,
+          _.get(temp, 'response.milestones.service_started') as string,
+          _.get(temp, 'response.service_name') as string
+        )
+    );
+    if (descriptions.length == 0) {
+      descriptions.push({
+        status: ExecJSON.ControlResultStatus.Passed,
+        code_desc: 'NA',
+        start_time: _.get(temp, 'response.milestones.service_started') as string
+      });
+    }
+    _.set(temp, 'result.sections', descriptions);
+    newout.push(temp);
+  });
+  const groups = _.groupBy(newout, (value) => {
+    return _.get(value, 'response.service_name');
+  });
+  return groups;
+}
+
+function controlMappingConveyor(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'sha256'},
+    title: {path: 'filename'},
+    desc: {path: 'cve_summary'},
+    impact: {
+      path: 'result.score',
+      transformer: (value) => {
+        return value / 1000;
+      }
+    },
+    refs: [],
+    tags: {
+      nist: DEFAULT_NIST_TAG
+    },
+    source_location: {
+      ref: ''
+    },
+    results: [
+      {
+        path: 'result.sections',
+        status: {path: 'status'},
+        code_desc: {path: 'code_desc'},
+        start_time: {path: 'start_time'}
+      }
+    ]
+  };
+}
+
+export class ConveyorMapper extends BaseConverter {
+  data: Record<string, unknown>;
+  type: string;
+  defaultMapping(): MappedTransform<ExecJSON.Execution, ILookupPath> {
+    return {
+      platform: {
+        name: 'Heimdall Tools',
+        release: HeimdallToolsVersion
+      },
+      version: HeimdallToolsVersion,
+      statistics: {},
+      profiles: [
+        {
+          name: this.type,
+          version: {path: 'api_server_version'},
+          title: {path: 'api_response.params.description'},
+          supports: [],
+          attributes: [],
+          groups: [],
+          status: 'loaded',
+          controls: [
+            {
+              path: 'api_response.results',
+              ...controlMappingConveyor()
+            }
+          ],
+          sha256: ''
+        }
+      ]
+    };
+  }
+  constructor(
+    conveyor: Record<string, unknown>,
+    meta: Record<string, unknown>,
+    type: string
+  ) {
+    const data = meta;
+    _.set(data, 'api_response.results', conveyor);
+    super(data);
+    this.type = type;
+    this.data = data;
+    this.setMappings(this.defaultMapping());
+  }
+}
+
+export class ConveyorResults {
+  data: unknown;
+  meta: Record<string, unknown>;
+  constructor(ConveyorJson: string) {
+    const data = JSON.parse(ConveyorJson);
+    const mapped = shafileMapper(data);
+    const newres = arrayifyObject(data, mapped);
+    this.data = newres;
+    this.meta = _.omit(data, 'api_response.results');
+  }
+
+  toHdf(): Record<string, ExecJSON.Execution> {
+    const mapped = _.mapValues(
```

Comment:
could be replaced by something that looks like the following.  it'll be pretty similar to what you already have here but using more native syntax instead of lodash.

toHdf(): {
return Object.entries(this.data).map(([key, val]) => return new ConveyorMapper(stuff).toHdf());
}

Reply:
this returns an array, does it still make sense to change it to this method, to then have to transform the result again?